### ISSUE TITLE
djpegli: Can't decompress an image

### ISSUE THREAD

**Issue by j7nj7n** [libjxl/libjxl#3570]
I found a JPEG image that cannot be decompressed by djpegli.exe. I can't say what exactly is wrong with it. It has unoptimal huffman encoding. It works after optimization, which removes restart marker 0xFFDD.

https://coverartarchive.org/release/9d628829-6a1c-47b5-9d72-3f0be929cd55/25011158016.jpg

> t:\working>djpegli.exe "25011158016.jpg" "25011158016.pnm"
> Read 2734380 compressed bytes.
> jpegli decoding failed

**Comment by mo271** [comment]
Thanks for the report. I'm having trouble to repro this. 
It seems to work just fine at current head:

```
> ./build/tools/djpegli 25011158016.jpg out.pnm
Read 2734380 compressed bytes.
1425 x 1400,  2.696 MP/s [2.70, 2.70], , 1 reps, 1 threads.
lib/extras/enc/pnm.cc:38: PNM encoder ignoring metadata - use a different codec
```

What version/commit did you use?

**Comment by mo271** [comment]
bisecting this: seems like it was fixed by #3503, please confirm.

**Comment by j7nj7n** [comment]
I have the last release v0.10.2. I can't compile a later version myself. Yes, the quantization table comes late in this file. Good to know that it is fixed, and this issue can be closed.
